lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -115,13 +115,16 @@ def self.process_result(ae_result, aevent)
   end
 
   def self.call_automate(obj, attrs, instance_name, options = {})
-    user = obj.tenant_identity
+    user = User.current_user || obj.tenant_identity
```

Comment:
@lfu @gmcculloug 
When these events are raised and if the user is part of multiple groups which group should get used. Do we record the group with the events or vm ownership?

Reply:
@mkanoor The user and the group is recorded in the Queue when the event is raised. Then the user is set to the group in the memory when the item get dequeued and processed.